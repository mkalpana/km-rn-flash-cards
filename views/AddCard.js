import React, { Component } from 'react';
import { StyleSheet, KeyboardAvoidingView } from 'react-native';
import PropTypes from 'prop-types';
import { Button, Input } from '../components';
import { addCard, fetchDecks } from '../actions';
import { connect } from 'react-redux';
import { get } from 'lodash';

class AddCard extends Component{
  state = {
    title: '',
    question: '',
    answer: ''
  };

  _onAddCard = () => {
    const { question, answer } = this.state;
    const { title } = this.props;
    this.props.addCard(title, { question, answer })
      .then(() => {
        this.props.fetchDecks();
        this.setState({ question: '', answer: '' });
        this.props.navigation.navigate('Deck', { title });
      });
  };

  render() {
    const { question, answer } = this.state;
    return (
      <KeyboardAvoidingView style={styles.container} behavior="padding">
        <Input
          style={styles.inputs}
          placeholder="Enter question"
          value={this.state.question}
          onChangeText={(question) => this.setState({ question })}
        />
        <Input
          style={styles.inputs}
          placeholder="Enter answer"
          value={this.state.answer}
          onChangeText={(answer) => this.setState({ answer })}
        />
        <Button
          style={styles.btnStyle}
          label="Submit"
          disabled={!question || !answer}
          onPress={this._onAddCard} />
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    margin: 10,
  },
  inputs: {
    marginTop: 30,
    marginBottom: 15,
    marginLeft: 20,
    marginRight: 20,
  },
  btnStyle: {
    marginTop: 40,
    marginBottom: 10,
    marginLeft: 20,
    marginRight: 20
  }
});

AddCard.propTypes = {
  title: PropTypes.string.isRequired,
  addCard: PropTypes.func.isRequired,
  fetchDecks: PropTypes.func.isRequired,
};

AddCard.defaultPropTypes = {
  title: ''
};

const mapStateToProps = (state, ownProps) => {
  const { navigation } = ownProps;
  const title = get(navigation, 'state.params.title', '');
  return {
    title
  };
};

export default connect(mapStateToProps, {
  addCard,
  fetchDecks
})(AddCard);
