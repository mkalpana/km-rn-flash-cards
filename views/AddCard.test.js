import React from 'react';
import { AddCard } from './AddCard';
import renderer from 'react-test-renderer';

describe('AddCard', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<AddCard
      title="React"
      addCard={() => ({})}
      fetchDecks={() => ([])}
    />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
