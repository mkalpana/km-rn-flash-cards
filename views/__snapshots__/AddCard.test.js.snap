// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard renders without crashing 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "margin": 10,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter question"
    required={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontSize": 16,
          "marginBottom": 10,
          "padding": 10,
        },
        Object {
          "marginBottom": 15,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 30,
        },
      ]
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter answer"
    required={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontSize": 16,
          "marginBottom": 10,
          "padding": 10,
        },
        Object {
          "marginBottom": 15,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 30,
        },
      ]
    }
    value=""
  />
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 40,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 5,
              "borderWidth": 2,
              "padding": 20,
              "paddingLeft": 40,
              "paddingRight": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
